pkg/scheduler/ugm/user_tracker_test.go

```diff
@@ -224,6 +222,17 @@ func TestSetMaxLimits(t *testing.T) {
 	userTracker.setLimits(hierarchy5, usage1, 1)
 }
 
+func TestSetGroupForApp(t *testing.T) {
+	userTracker := newUserTracker("user")
+	groupTracker1 := newGroupTracker("group1")
+	groupTracker2 := newGroupTracker("group2")
+	userTracker.setGroupForApp(TestApp1, groupTracker1)
+	userTracker.setGroupForApp(TestApp2, groupTracker2)
+
+	assert.Equal(t, 2, len(userTracker.getTrackedApplications()))
+	assert.Equal(t, 2, len(userTracker.appGroupTrackers))
+}
```

Comment:
Do we really need this new test? I think we can move these two asserts after line no. 76 in earlier method?

Reply:
My original thought was to add unit test that can partially address the previous todo comment. 
//nolint: todo test cases for different applications each with different group linkage

Just removed it to simplify this PR.